packages/electron-updater/src/RpmUpdater.ts

```diff
@@ -32,26 +32,31 @@ export class RpmUpdater extends BaseUpdater {
   }
 
   protected doInstall(options: InstallOptions): boolean {
-    const sudo = this.wrapSudo()
-    // pkexec doesn't want the command to be wrapped in " quotes
-    const wrapper = /pkexec/i.test(sudo) ? "" : `"`
-    const packageManager = this.spawnSyncLog("which zypper")
-    const installerPath = this.installerPath
-    if (installerPath == null) {
-      this.dispatchError(new Error("No valid update available, can't quit and install"))
+    try {
+      const sudo = this.wrapSudo()
+      // pkexec doesn't want the command to be wrapped in " quotes
+      const wrapper = /pkexec/i.test(sudo) ? "" : `"`
+      const packageManager = this.spawnSyncLog("which zypper")
+      const installerPath = this.installerPath
+      if (installerPath == null) {
+        this.dispatchError(new Error("No valid update available, can't quit and install"))
+        return false
+      }
+      let cmd: string[]
+      if (!packageManager) {
+        const packageManager = this.spawnSyncLog("which dnf || which yum")
+        cmd = [packageManager, "-y", "install", installerPath]
+      } else {
+        cmd = [packageManager, "--no-refresh", "install", "--allow-unsigned-rpm", "-y", "-f", installerPath]
+      }
+      this.spawnSyncLog(sudo, [`${wrapper}/bin/bash`, "-c", `'${cmd.join(" ")}'${wrapper}`])
+      if (options.isForceRunAfter) {
+        this.app.relaunch()
+      }
+      return true
+    } catch (e) {
```

Comment:
Looks like we don't need to wrap this with try-catch as it's already done on in the call to `doInstall`
https://github.com/electron-userland/electron-builder/blob/bee179b3cf8163041d280ed8dc5a5ce4f27786c6/packages/electron-updater/src/BaseUpdater.ts#L65-L75

I'll push an update to this PR